### PR TITLE
Bug 2088660: Fix for absence folder for bootstrap ignition iso

### PR DESCRIPTION
**Pull request by adiantum**
- ensure that folder exists before file creation

### CODE REVIEW

File: pkg/types/nutanix/helpers.go [hunk 1]
```diff
@@ -78,6 +78,23 @@ func CreateBootstrapISO(infraID, userData string) (string, error) {
 	}
 
 	fullISOFile := BootISOImagePath(cacheDir, infraID)
+	fullISOFileDir, err := filepath.Abs(filepath.Dir(fullISOFile))
```
Comment (rvanderp3) [inline pkg/types/nutanix/helpers.go:81]: Is the call to filepath.Abs necessary here?  [Abs](https://pkg.go.dev/path/filepath#Abs) will set the provided path relative to the current working directory if a full path isn't provided and I'm not sure that was the intention here.  It looks like UserCacheDir already provides an absolute path that is consumed by BootISOImagePath.  I'm not sure the call to Abs is necessary here.  I could be wrong on that.

Comment (adiantum) [reply pkg/types/nutanix/helpers.go:81]: Unfortunately, there is no guarantee that os.UserCacheDir() will return absolute path: https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/os/file.go;l=406

I've checked it at least on Centos 7, it's possible to have conditions when the function return relative path.

**Pull request merged**